solarforecastarbiter/reference_forecasts/persistence.py

```diff
@@ -362,3 +365,244 @@ def _check_intervals_times(interval_label, data_start, data_end,
+        Forecast interval length
+    interval_label : str
+        instant, beginning, or ending
+    load_data : function
+        A function that loads the observation data. Must have the
+        signature load_data(observation, data_start, data_end) and
+        properly account for observation interval label.
+    axis : {'x', 'y'}
+        The axis on whicht he constant values of the CDF is specified. The axis
+        can be either *x* (constant variable values) or *y* (constant
+        percentiles).
+    constant_values : array_like
+        The variable values or percentiles.
+
+    Returns
+    -------
+    forecasts : list of pd.Series
+        The persistence forecasts, returned in the same order as
+        *constant_values*. If *axis* is *x*, the forecast values are
+        percentiles (e.g. 25%). If instead *axis* is *y*, the forecasts values
+        have the same units as the observation data (e.g. MW).
+
+    Raises
+    ------
+    ValueError
+        If there is insufficient data for matching by time of day or the
+        **axis** parameter is invalid.
+
+    Notes
+    -----
+    Assumes that there is at least 20 days of *observation* data available
+    based on [1]_, [2]_, [3]_.
+
+    References
+    ----------
+    .. [1] Allessandrini et al. (2015) "An analog ensemble for short-term
+       probabilistic solar power forecast", Appl. Energy 157, pp. 95-110.
+       doi: 10.1016/j.apenergy.2015.08.011
+    .. [2] Yang (2019) "A universal benchmarking method for probabilistic
+       solar irradiance forecasting", Solar Energy 184, pp. 410-416.
+       doi: 10.1016/j.solener.2019.04.018
+    .. [3] Doubleday, Van Scyoc Herndandez and Hodge (2020) "Benchmark
+       probabilistic solar forecasts: characteristics and recommendations",
+       Solar Energy 206, pp. 52-67. doi: 10.1016/j.solener.2020.05.051
+
+    See also
+    --------
+    :py:func:`solarforecastarbiter.reference_forecasts.persistence.persistence_probabilistic`
+
+    """
+    # ensure that we're using times rounded to multiple of interval_length
+    _check_intervals_times(observation.interval_label, data_start, data_end,
+                           forecast_start, forecast_end,
+                           observation.interval_length)
+
+    closed = datamodel.CLOSED_MAPPING[interval_label]
+    fx_index = pd.date_range(start=forecast_start, end=forecast_end,
+                             freq=interval_length, closed=closed)
+
+    # observation data resampled to match forecast sampling
+    obs = load_data(observation, data_start, data_end)
+    obs = obs.resample(interval_length, closed=closed).mean()
+
+    # time of day: minutes past midnight (e.g. 0=12:00am, 75=1:15am)
+    obs_timeofday = (obs.index.hour * 60 + obs.index.minute).astype(int)
+    fx_timeofday = (fx_index.hour * 60 + fx_index.minute).astype(int)
+
+    # confirm sufficient data for matching by time of day
+    if data_end - data_start < pd.Timedelta('20D'):
```

Comment:
hmm, should we actually compare 
```python
obs.last_valid_index() - obs.first_valid_index() < pd.Timedelta('20D')
```

``load_data`` likely only fetches the available data between start and end but doesn't guarantee there is data at all

Reply:
Revised as suggested.